Mathlib/Tactic/Simpa.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Mario Carneiro
+-/
+import Mathlib.Tactic.Core
+
+namespace Mathlib.Tactic
+
+open  Lean.Parser.Tactic Lean.Elab.Tactic
+
+-- move these?
+syntax simpArgs' := ((" only ")? " [" simpArg,+ "] ")?
+syntax withStx   := (" with " (colGt ident)+)?
+syntax usingStx  := (" using " term)?
+
+elab (name := simpa) "simpa" cfg?:(config)? disch?:(discharger)?
+    args?:simpArgs' wth?:withStx using?:usingStx : tactic => do
+  let cfg := cfg?.getOptional?
+  let args := args?.getOptional?
+  let nGoals := (← getUnsolvedGoals).length
+  try evalTactic (← `(tactic|simp $(cfg)? $(disch?.getOptional?)? $(args)?))
+  catch | _ => throwError "couldn't simplify the goal"
```

Comment:
This is not how the current simpa behaves:
```
example (p : ℕ → Prop) (h : p (1 + 0)) : p 1 := by simpa -- works
```

Reply:
I didn't understand it. The following also works with my implementation:
```lean
example (p : Nat → Prop) (h : p (1 + 0)) : p 1 := by simpa
```

Reply:
Ah, I forgot that in Lean 4 simp doesn't fail if it doesn't simplify.